src/p2p/net_node.inl

```diff
@@ -2282,6 +2295,15 @@ namespace nodetool
     return true;
   }
   //-----------------------------------------------------------------------------------
+  epee::net_utils::network_address get_preferred_address(const epee::net_utils::connection_context_base& context, const basic_node_data& node_data)
+  {
+    epee::net_utils::network_address na = context.m_remote_address;
+    if (node_data.my_advertise_network_address.get_type_id() != epee::net_utils::address_type::invalid)  {
+      na = node_data.my_advertise_network_address;
```

Comment:
I think the check should be `if (context.m_remote_address.get_type_id() == node_data.my_advertise_network_address.get_type_id())` - otherwise someone could specify an external address from a different zone which should never be allowed.